packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -522,43 +478,50 @@ export default class API {
           // filter only files and up to the required depth
           .filter(file => file.type === 'blob' && file.path.split('/').length <= depth)
           .map(file => ({
-            ...file,
+            type: file.type,
+            id: file.sha,
             name: basename(file.path),
             path: `${folder}/${file.path}`,
+            size: file.size,
           })),
       )
       .catch(replace404WithEmptyArray);
   }
 
-  readUnpublishedBranchFile(contentKey: string) {
-    const metaDataPromise = this.retrieveMetadata(contentKey).then(data =>
-      data.objects.entry.path ? data : Promise.reject(null),
-    );
-    const repoURL = this.useOpenAuthoring
-      ? `/repos/${contentKey
-          .split('/')
-          .slice(0, 2)
-          .join('/')}`
-      : this.repoURL;
-    return resolvePromiseProperties({
-      metaData: metaDataPromise,
-      fileData: metaDataPromise.then(data =>
-        this.readFile(data.objects.entry.path, null, {
-          branch: data.branch,
+  async readUnpublishedBranchFile(contentKey: string) {
+    try {
+      const metaData = await this.retrieveMetadata(contentKey).then(data =>
+        data.objects.entry.path ? data : Promise.reject(null),
+      );
+      const repoURL = this.useOpenAuthoring
+        ? `/repos/${contentKey
+            .split('/')
+            .slice(0, 2)
+            .join('/')}`
+        : this.repoURL;
+
+      const [fileData, isModification] = await Promise.all([
```

Comment:
I find this easier to read instead of the nesting when using `resolvePromiseProperties `